Sources/Core/Extensions/Foundation/Array+PovioKit.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  Array+PovioKit.swift
+//  PovioKit
+//
+//  Created by Borut Tomazin on 02/10/2024.
+//  Copyright © 2024 Povio Inc. All rights reserved.
+//
+
+import Foundation
+
+extension Array {
+  /// Splits the array into chunks of a specified size.
+  ///
+  /// - Parameter size: The size of each chunk. Each chunk will contain up to `size` elements.
+  /// - Returns: An array of arrays, where each inner array is a chunk of the original array.
+  ///
+  /// - Note: If the array's size is not a perfect multiple of `size`, the last chunk will contain the remaining elements.
+  ///
+  /// ## Example
+  /// ```
+  /// let array = [1, 2, 3, 4, 5, 6, 7]
+  /// let chunks = array.chunked(into: 3)
+  /// // chunks is [[1, 2, 3], [4, 5, 6], [7]]
+  /// ```
+  func chunked(into size: Int) -> [[Element]] {
+    stride(from: 0, to: count, by: size).map {
+      Array(self[$0 ..< Swift.min($0 + size, count)])
+    }
```

Comment:
It would be much faster to return a list of subsequences.

Reply:
We can also make this method more general by extending `Collection`.

Reply:
RandomAccessCollection is indeed the best choice 👌